chore(server): avoid copying sources in dev

yonran opened the pull request:
Add a dev target to the web and server Dockerfiles, and change docker-compose.dev.yml to use the dev target. The dev target avoids copying files so that the docker image is smaller.

My motivation was that when I tried using docker-compose.dev.yml, I kept running out of disk space because docker would create big docker containers with copies of the application code even though they weren’t being used.

yonran committed:
chore(server): avoid copying sources in dev

Add a dev target to the web and server Dockerfiles, and change docker-compose.dev.yml to use the dev target. The dev target avoids copying files so that the docker image is smaller.

jrasm91 commented on docker/docker-compose.dev.yml:
```diff
@@ -55,6 +55,7 @@ services:
     image: immich-web-dev:latest
     build:
       context: ../web
+      target: dev
```
the web Dockerfile is only ever used for development so you can revert related changes.

yonran replied:
Sorry for taking so long. I have removed the `target: dev` and removed the `COPY` from the `web/Dockerfile` completely so that it can only be used with the source volume.

alextran1502 commented:
Hello, any update on this PR?

yonran committed:
chore: respond to PR: don't add dev target

web/Dockerfile is only used by docker-compose.dev.yml so a dev target is redundant. Instead, just remove the copy

zackpollard commented:
Hey @yonran, is this ready to be reviewed and merged? Changes look simple, if they're good from your end I can give this a quick test and merge it in.

jrasm91 committed:
Merge branch 'main' into docker-compose-dev-faster
